src/cpp/utils.cpp

```diff
@@ -0,0 +1,33 @@
+#include "utils.h"
+
+#include <fstream>
+#include <iomanip>
+
+namespace effil {
+
+Logger getLogger() {
+    static std::mutex lock;
```

Comment:
Logger should own mutex.

Reply:
currently `Logger` is just scoped wrapper, if we put `mutex` and `stream` inside `Logger` class we will have to make another scoped wrapper to restrict the mutex locking time. Do you think it's good idea?